libraries/AP_HAL_ChibiOS/I2CDevice.cpp

```diff
@@ -105,6 +105,45 @@ void I2CBus::clear_all()
     }
 }
 
+/*
+  If bus exists, set its data and clock lines to floating
+ */
+void I2CBus::set_bus_to_floating(uint8_t busidx)
+{
+    if (busidx < ARRAY_SIZE(I2CD)) {
+        const struct I2CInfo &info = I2CD[busidx];
+        const ioline_t sda_line = GPIO::resolve_alt_config(info.sda_line, PERIPH_TYPE::I2C_SDA, info.instance);
+        const ioline_t scl_line = GPIO::resolve_alt_config(info.scl_line, PERIPH_TYPE::I2C_SCL, info.instance);
+        palSetLineMode(sda_line, PAL_MODE_INPUT);
+        palSetLineMode(scl_line, PAL_MODE_INPUT);
+    }
+}
+
+
+/*
+  Check enabled I2C/CAN select pins against check_pins bitmask
+ */
+uint8_t I2CBus::check_select_pins(uint8_t check_pins)
```

Comment:
This function outputs an `uint8_t`, but you have a boolean output at the return being converted

```suggestion
bool I2CBus::check_select_pins(uint8_t check_pins)
```

Reply:
edited changes in by rebasing